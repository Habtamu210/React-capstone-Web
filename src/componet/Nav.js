import React from 'react';
import { NavLink } from 'react-router-dom';

const Nav = () => (
  <nav className="nav bg-danger ">
    <ul className="d-flex justify-content-between w-100 my-2 mx-0">
      <li className="fonts d-flex gap-2 list-unstyled">
        <NavLink
          className="text-decoration-none"
          to="/"
        >
          <h3 className="fonts"> ‚áê</h3>
        </NavLink>
        <h5 className="pl-3 mt-0 fonts pad font-weight-light">Crypto Coins Daily Excnagehe Value</h5>
      </li>
      <li className="d-flex list-unstyled px-0">
        <h1 className="px-5 font-weight-light">&#937;</h1>
      </li>
    </ul>
  </nav>
);

export default Nav;
