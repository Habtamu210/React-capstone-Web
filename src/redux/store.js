import { configureStore } from '@reduxjs/toolkit';
import cryptoReducer from './Model';

const store = configureStore({
  reducer: {
    crypto: cryptoReducer,
  },
});
export default store;