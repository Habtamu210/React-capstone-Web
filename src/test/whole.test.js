import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Nav from '../componet/Nav';
import Models from '../componet/Render';
import Crypto from '../componet/defCrypto';

const crypto = { FULLName: 'bitcoin', img: 'new' };
jest.mock('axios');

describe('Nav component', () => {
  it('renders Nav component', () => {
    const r = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(r).toMatchSnapshot();
  });
  it('renders Models component', () => {
    const m = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Models />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(m).toMatchSnapshot();
  });
  it('renders Crypto component', () => {
    const m = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Crypto crypto={crypto} />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(m).toMatchSnapshot();
  });
});
