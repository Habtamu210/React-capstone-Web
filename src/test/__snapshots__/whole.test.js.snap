// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders Crypto component 1`] = `
<div
  className="rounded-4 w-100 h-100 border border-danger d-flex justify-content-center"
>
  <div
    className="w-100"
  >
    <p
      className=" p-3 font text-light"
    >
       
      <span>
        Name:
      </span>
    </p>
    <div
      className="d-flex m-4 justify-content-center"
    >
      <img
        alt="symbol"
        className="images w-70"
        src="https://www.cryptocompare.comnew "
      />
    </div>
    <p />
  </div>
</div>
`;

exports[`Nav component renders Models component 1`] = `
Array [
  <form
    className="w-100 mb-5 bg-secondary nav"
  >
    <input
      aria-label="search"
      className="input w-100 h-2"
      onChange={[Function]}
      placeholder=" Search daily crypto exchange values"
      type="text"
      value=""
    />
  </form>,
  <div
    className="bg-primary text-white"
  >
    <div
      className="grid1"
    />
  </div>,
]
`;

exports[`Nav component renders Nav component 1`] = `
<nav
  className="nav bg-danger "
>
  <ul
    className="d-flex justify-content-between w-100 my-2 mx-0"
  >
    <li
      className="fonts d-flex gap-2 list-unstyled"
    >
      <a
        aria-current="page"
        className="text-decoration-none active"
        href="/"
        onClick={[Function]}
      >
        <h3
          className="fonts"
        >
           ⇐
        </h3>
      </a>
      <h5
        className="pl-3 mt-0 fonts pad font-weight-light"
      >
        Crypto Coins Daily Excnagehe Value
      </h5>
    </li>
    <li
      className="d-flex list-unstyled px-0"
    >
      <h1
        className="px-5 font-weight-light"
      >
        Ω
      </h1>
    </li>
  </ul>
</nav>
`;
